tensorflow_lite_support/ios/Sources/TFLCommon.h

```diff
@@ -0,0 +1,211 @@
+/* Copyright 2021 The TensorFlow Authors. All Rights Reserved.
+ 
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+ 
+ http://www.apache.org/licenses/LICENSE-2.0
+ 
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ ==============================================================================*/
+
+#import <Foundation/Foundation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+/**
+ * @enum TFLSupportErrorCode
```

Comment:
AFAIK, ARC won't work on C structs. Hence we have to free C structs we use within our Obj C framework. If we reuse C structs, ObjC Client code will also have to take care of freeing them once allocated inspite off enabling ARC. Plus syntax for initialising, pointer reference etc for C data structures is non native to Obj C. You will still have to use the C syntax. iOS devs are not familiar with these. Hence we are hiding all these by way of creating native ObjC types over them. I have never called C funcs from Swift, so I am not sure if the same holds true in case of Swift client code as well.
 Enum `TfLiteSupportErrorCode` probably can be reused. But we loose the capability of defining things like `NS_SWIFT_NAME` for it. Also I have removed a lot of documentation that is specific to only C users of the library and some values that were defined for easy looping through the enum which are not needed in the ObjC library. Documentation maybe confusing if it's reused. I can investigate the enum reusability if need be.

Reply:
+1 on redefining the error code as ObjC's enum and structs as ObjC objects as it'll be easier for Swift interop. The API will be very hard for use for iOS developers if we reuse the C structs.